core/DataAccess/Model.php

```diff
@@ -149,6 +156,45 @@ public function updateArchiveAsInvalidated($archiveTable, $idSites, $datesByPeri
         return Db::query($sql, $bind);
     }
 
+    /**
+     * @param string $archiveTable Prefixed table name
+     * @param int[] $idSites
+     * @param Segment $segment
+     * @param array $datesByPeriodType
+     * @return \Zend_Db_Statement
+     * @throws Exception
+     */
+    public function updateAllRangeArchivesAsInvalidated($archiveTable, $idSites, $ranges, Segment $segment = null)
```

Comment:
Looking at the method `updateArchiveAsInvalidated` https://github.com/matomo-org/matomo/blob/3.x-dev/core/DataAccess/Model.php#L108-L119 I think this is actually updating it as expected using `(date1 <= ? AND ? <= date2)`? I wonder is this method even needed? Not sure how it behaves different to the other method?

Reply:
Actually realising there is a new `if` in the mentioned method that only deletes an exact range when `--periods=range` is set. Not sure we want that behaviour. I reckon we can remove that `if` and restore old behaviour and then we can also use the `updateArchiveAsInvalidated ` method again. I reckon it might be fine to use same behaviour for when using `--periods=range` and when using `--periods=all`. Currently it seems to be using different behaviour when using `--periods=range` where it only deletes the exact given range but it should maybe delete all ranges it is present.

Reply:
ping @mattab not exactly sure what behaviour we want

Reply:
> using --periods=all. Currently it seems to be using different behaviour when using --periods=range where it only deletes the exact given range but it should maybe delete all ranges it is present.

:+1:  when `--periods=all` then all ranges should be invalidated.